src/core.rs

```diff
@@ -629,3 +684,15 @@ impl Mats {
         self.inner.length
     }
 }
+
+/// Split creates an array of single channel images from a multi-channel image
+pub fn split(src: &Mat) -> Mats {
+    let total_length = src.channels() as usize;
+    let mut mats_: Vec<*mut ::std::os::raw::c_void> = Vec::with_capacity(total_length);
```

Comment:
You're going to need to use Vec::as_boxed_slice to avoid a memory leak here.

Reply:
created new_of_len impl